pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -71,3 +102,230 @@ func (r *PerformanceProfileReconciler) hypershiftSetupWithManager(mgr ctrl.Manag
+		if k8serrors.IsNotFound(err) {
+			//FIXME - In hypershift there is no "owned" objects. So how do we delete created objects? or do we even have to?
+			//jlom - as we create a configmap for each of the elements created from a PerformanceProfile and we reference
+			//       these configmaps in NodePool so they could be "transfered" to the managed cluster, maybe "delete" them is
+			//       as simple as derreference them in NodePool and delete the ConfigMaps.
+			//NOTE - Adding a label with PerformanceProfile name on it to the configmaps could make this look up easier.
+
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	performanceProfileString, ok := instance.Data[tunedConfigMapConfigKey]
+	if !ok {
+		klog.Warning("ConfigMap %s has no data for field %s", instance.ObjectMeta.Name, tunedConfigMapConfigKey)
+
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	cmNodePoolNamespacedName, ok := instance.Annotations[hypershiftNodePoolLabel]
+	if !ok {
+		klog.Warningf("failed to parse PerformanceProfileManifests in ConfigMap %s, no label %s", instance.ObjectMeta.Name, hypershiftNodePoolLabel)
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+	nodePoolName := parseNamespacedName(cmNodePoolNamespacedName)
+
+	performanceProfileFromConfigMap, err := parsePerformanceProfileManifest([]byte(performanceProfileString), nodePoolName)
+	if err != nil {
+		klog.Warningf("failed to parseTunedManifests in ConfigMap %s: %v", instance.ObjectMeta.Name, err)
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	pinningMode, err := r.getInfraPartitioningMode()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	//REVIEW - jlom Not fully sure if `getContinerRuntimeName` gonna work in hypershift env
+	ctrRuntime, err := r.getContainerRuntimeName(ctx, performanceProfileFromConfigMap)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("hypershift could not determine high-performance runtime class container-runtime for profile %q; %w", performanceProfileFromConfigMap.Name, err)
+	}
+	klog.Infof("hypershift using %q as high-performance runtime class container-runtime for profile %q", ctrRuntime, performanceProfileFromConfigMap.Name)
+
+	componentSet, err := manifestset.GetNewComponents(performanceProfileFromConfigMap, nil, &pinningMode, ctrRuntime)
+	if err != nil {
+		klog.Errorf("failed to deploy performance profile from configMap %q components: %v", instance.Name, err)
+		//REVIEW - Should we record this events? and if so where? in the CM or the PP?
+		return reconcile.Result{}, err
+	}
+
+	//jlom - Now we have to create a ConfigMap for each of the elements in the componentSet and then handle them to
```

Comment:
no need to prefix clarifying comment with the author (git blame will tell us anyway if we're curious). If it's a development comment, though, let's remove it